src/lib/classes/saml_admin.php

```diff
@@ -22,43 +22,72 @@ function admin_menus()
   		add_action('admin_menu', array($this,'saml_sp_menus'));
   	}
   }
-  
+
   function saml_idp_menus()
   {
   	if( is_multisite() )
   	{
   		add_submenu_page('settings.php', 'Single Sign-On', 'Single Sign-On', 'manage_network', 'sso_idp.php', array($this,'sso_idp'));
   		add_submenu_page('settings.php', 'Single Sign-On', 'Single Sign-On', 'manage_network', 'sso_help.php', array($this,'sso_help'));
-  		
+
   		remove_submenu_page( 'settings.php', 'sso_help.php' );
   	}
   	else
   	{
   		add_submenu_page('options-general.php', 'Single Sign-On', 'Single Sign-On', 'administrator', 'sso_idp.php', array($this,'sso_idp'));
   		add_submenu_page('options-general.php', 'Single Sign-On', 'Single Sign-On', 'administrator', 'sso_help.php', array($this,'sso_help'));
-  		
+
   		remove_submenu_page( 'options-general.php', 'sso_idp.php' );
   		remove_submenu_page( 'options-general.php', 'sso_help.php' );
   	}
   }
-  
+
   function saml_sp_menus()
   {
   	add_submenu_page('options-general.php', 'Single Sign-On', 'Single Sign-On', 'administrator', 'sso_general.php', array($this,'sso_general'));
   	add_submenu_page('options-general.php', 'Single Sign-On', 'Single Sign-On', 'administrator', 'sso_sp.php', array($this,'sso_sp'));
   	add_submenu_page('options-general.php', 'Single Sign-On', 'Single Sign-On', 'administrator', 'sso_help.php', array($this,'sso_help'));
-  	
+
   	remove_submenu_page( 'options-general.php', 'sso_sp.php');
   	remove_submenu_page( 'options-general.php', 'sso_help.php');
   }
-  
+
+  /**
+   * Get identity provider details
+   * @return array
+   */
+  private function _get_idp_details()
+  {
+      $config_path = constant('SAMLAUTH_CONF') . '/config/saml20-idp-remote.ini';
+      $idp_details = null;
+      $idp_settings = $this->settings->get_idp_details();
+
+      /*
+       Read configuration details from database, if none is available,
+       check for a configuration file, otherwise use default values.
+       */
+      if($idp_settings) {
+          $idp_details = parse_ini_string($idp_settings, true);
+      }
+      elseif(file_exists($config_path))
+      {
+          $idp_details = parse_ini_file($config_path, true);
+      }
+      else
+      {
+          $idp_details = parse_ini_string("[https://your-idp.net]\nname = Your IdP\nSingleSignOnService = https://your-idp.net/SSOService\nSingleLogoutService = https://your-idp.net/SingleLogoutService\ncertFingerprint = 0000000000000000000000000000000000000000", true);
```

Comment:
I don't like the idea of continuing to use the INI format when an actual INI file is no longer being used. I think it would be more natural to represent the IdP data as a PHP array:

```php
$idp_details = array (
  'https://your-idp.net' => 
  array (
    'name' => 'Your IdP',
    'SingleSignOnService' => 'https://your-idp.net/SSOService',
    'SingleLogoutService' => 'https://your-idp.net/SingleLogoutService',
    'certFingerprint' => '0000000000000000000000000000000000000000',
  ),
);
```

This will automatically be serialized when written to the `wp_options` table and deserialized when read, so we'd be able to remove all the code that tries to do INI serialization and make reading and updating the IdP data a lot simpler.